### PR TITLE
Closes #849 create adpc template@devel

### LINKED ISSUE #849: Create ADPC

### PR DESCRIPTION
**Pull request by jeffreyad**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by jeffreyad** [comment]
Replaces #1432

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:64]:
```suggestion
# operates on vectors, which can be used in `mutate`.
```

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:45]: Should this format issue be fixed at the `admiral.test` level?

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:64]:
```suggestion
# Here is an example of how you can create your own function that
# operates on vectors, which can be used in `mutate`.
```

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:119]: Can we add this note to t Lines 87-93 as well?

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:182]: Should we add a note that all these times are 00:00:00 as our DTC variable was just a date and we did imputation.

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:201]: ohh can we explain what this is doing here?!?

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:321]:
```suggestion
    ),
    # Derive Reference Dose Date
```

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:408]: I fee like a lot is going on here.  The top-level comment  "Derive Analysis Variables AVAL, ATPT, ATPTREF and BASETYPE" is confusing as we do more then just those four in this section.  Can this be made clearer please?

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:456]: Can these just get collapsed into one section since they are pretty straightforward.

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:59]: I think this label is already on this dataset??  I see it when I run data("admiral_pc")
image

**Review by bms63 — CHANGES_REQUESTED** [review]
Very cool work!  I wish there was a spec so we could organize the dataset to be easier to read at the end.  Maybe worthwhile to create something?!? 

@thomas-neitmann Can you look over this template as well please

**Review by bms63 — COMMENTED** [review]
@jeffreyad we also need a reference in the news file for this dataset template please.

File: inst/templates/ad_adpc.R
Comment (jeffreyad) [reply inst/templates/ad_adpc.R]: Yes. Removed this formatting code.  It looks like the issue is resolved.  If there is still an issue, it should be fixed at `admiral.test` level.

File: inst/templates/ad_adpc.R
Comment (jeffreyad) [reply inst/templates/ad_adpc.R]: Yes, updated comments to clarify.

File: inst/templates/ad_adpc.R
Comment (jeffreyad) [reply inst/templates/ad_adpc.R:180]: Yes, I think this is clearer now with additional comments.

File: inst/templates/ad_adpc.R
Comment (jeffreyad) [reply inst/templates/ad_adpc.R:201]: Yes, added additional comments to clarify.

File: inst/templates/ad_adpc.R
Comment (jeffreyad) [reply inst/templates/ad_adpc.R]: Yes, added additional comments.

File: inst/templates/ad_adpc.R
Comment (jeffreyad) [reply inst/templates/ad_adpc.R]: Done.

**Comment by jeffreyad** [comment]
> Very cool work! I wish there was a spec so we could organize the dataset to be easier to read at the end. Maybe worthwhile to create something?!?
> 
> @thomas-neitmann Can you look over this template as well please

@bms63 I could add a spec like you did in the Hackathon demo.  I have a working version of this which I could potentially add.

**Comment by bms63** [comment]
I would love that as the pk dataset is kind of hard to look at - especially as I just want to look at the stuff derived in the dataset and not the adsl variables..  I wonder how much a simple spec file would add to the dataload of the package??

@pharmaverse/admiral what do you think of adding a spec file for at least this dataset?  Maybe others?

Honestly, I think it would really allow us to have better presentation of the final dataset in our templates and in our vignettes.  They are very hard to look at, but I am worried about storage and maintainability and how we would apply them.  Perhaps there is some way to store them on GitHub AWS server and query them from there??  Code could be commented out so we don't make `metatools`, `xportr` other packages depenedencies.

File: inst/templates/ad_adpc.R
Comment (bms63) [inline inst/templates/ad_adpc.R:21]: Hey @jeffreyad.  I really want this to happen, but it does add a lot of dependencies for the user.  I will bring this up to the admiral core team on Monday/Wednesday to discuss.  But this will be a separate issue for making this update. 

Can you store this on box for us to use later and revert your changes.  After that I think we can start merging in all these PK/PC materials you have been working on.

Comment (jeffreyad) [reply inst/templates/ad_adpc.R:21]: @bms63 Sure, will do.  Thanks

**Review by bms63 — APPROVED** [review]

**Comment by bms63** [comment]
omg!  we are getting close!  Thanks again @jeffreyad for  carrying this through!!

**Pull request merged**
